search_engine_parser/core/utils.py

```diff
@@ -53,8 +69,13 @@ async def get_source(self, engine, url, headers, cache=True):
         if os.path.exists(cache_path) and cache:
             with open(cache_path, 'rb') as stream:
                 return pickle.load(stream), True
+        get_vars = { 'url':url, 'headers':headers }
+        if proxy and proxy_auth:
```

Comment:
For randomness sakes,  I don't know if we should make proxy apply sometimes and ignored sometimes.  Or in use cases where proxies should be used if the search hits a captcha

Reply:
If proxy isn't sent it isn't used. I don't think it's a good idea to provide the API and underneath make it random. If you don't want proxy, don't use proxy else use it

Reply:
Great